calico_containers/pycalico/ipam.py

```diff
@@ -85,12 +85,17 @@ def assign_address(self, pool, address):
         The directory for storing assignments in this pool must already exist.
 
         :param IPPool or IPNetwork pool: The pool that the assignment is from.
+        If pool is None, get the pool from datastore
         :param IPAddress address: The address to assign.
-
         :return: True if the allocation succeeds, false otherwise. An
         exception is thrown for any error conditions.
         :rtype: bool
         """
+        pool = pool or self.get_pool(address)
+        if pool is None:
+            raise KeyError("IP address %s does not belong to any configured"
```

Comment:
Instead of `KeyError`, we might want to define an API error for these sorts of things.

`InvalidPoolError` or something like that?  @robbrockbank any thoughts?